### PR TITLE
Move <threadapi> definition to Boost.Build

### PR DESCRIPTION
**Pull request by karzhenkov**
This is last step of moving `<threadapi>` and its auto-detection rule to `Boost.Build`.
Previous PRs are boostorg/build#251 and boostorg/boost#166.

### CODE REVIEW

**Comment by viboes** [comment]
But that this mean that the interface has changed now and that the user needs to use `threadapi-feature`?

File: build/Jamfile.v2 [hunk 1]
```diff
@@ -152,7 +152,7 @@ rule tag ( name : type ? : property-set )
         local api = [ $(property-set).get <threadapi> ] ;
```
Comment (viboes) [inline build/Jamfile.v2]: Why this is not changed?
Is to preserve the use of threadapi?

**Comment by karzhenkov** [comment]
The module `threadapi-feature.jam` is new, so there is no problem. The name of feature it defines is still `<threadapi>`.

**Comment by viboes** [comment]
Ok.

It is not working yet.

https://ci.appveyor.com/project/viboes/thread/build/1.0.134-develop/job/tj7us66lyvxkbctl

**Comment by viboes** [comment]
I've applied the patches locally and the build works on my macos.
However the test doesn't work.

There are some changes on test/Jamfile.v2 that I have not.

**Comment by viboes** [comment]
It is better after doing the cleanup as described here test/Jamfile.v2
https://github.com/boostorg/thread/commit/f67e6d44171c30df0819ab62238b0a18941baea1#diff-e97299ad55a4013dc19889043aa464a5

**Comment by karzhenkov** [comment]
test/Jamfile.v2 is the same in boostorg/thread@f67e6d4 and boostorg/thread@b128586, isn't it?

Did you add `<conditional>` to `Jamroot`?

**Comment by viboes** [comment]
Yes, I did locally.

**Comment by karzhenkov** [comment]
I can't reproduce the error you got. With clean checkout of top-level `develop` branch (boostorg/boost@3f32a98) test is started normally. There is no message "error: No best alternative for libs/thread/build/thread_sources". Could you check again?

**Comment by viboes** [comment]
I said "It is better after doing the cleanup as described here test/Jamfile.v2"

**Comment by karzhenkov** [comment]
`test/Jamfile.v2` in `develop` is reverted to state it has before #160. The [cleanup](https://github.com/boostorg/thread/commit/f67e6d44171c30df0819ab62238b0a18941baea1#diff-e97299ad55a4013dc19889043aa464a5) you point to does actually this rollback.

Did I miss something?
Is there still the test error?

**Comment by viboes** [comment]
Everything is ok :)

Thanks for working on this.

**Pull request merged**